dotnet/src/SemanticKernel.Abstractions/Kernel.cs

```diff
@@ -212,6 +222,28 @@ internal bool OnFunctionInvoked(FunctionInvokedEventArgs eventArgs)
         }
         return handled;
     }
+
+    internal bool OnPromptRendering(PromptRenderingEventArgs eventArgs)
+    {
+        bool handled = false;
+        if (this.PromptRendering != null)
+        {
+            this.PromptRendering.Invoke(this, eventArgs);
+            handled = true;
+        }
+        return handled;
+    }
+
+    internal bool OnPromptRendered(PromptRenderedEventArgs eventArgs)
+    {
+        bool handled = false;
+        if (this.PromptRendered != null)
+        {
+            this.PromptRendered.Invoke(this, eventArgs);
+            handled = true;
+        }
+        return handled;
+    }
```

Comment:
bool return is not being used, this method can be simplified to
```suggestion
    internal void OnPromptRendered(PromptRenderedEventArgs eventArgs)
    {
        this.PromptRendered?.Invoke(this, eventArgs);
    }
```

Same for `OnPromptRendering`

Reply:
@RogerBarreto I used the bool return value so that I know when to create a new `FunctionResult` as I don't wnat to do that unless needed.

```c#
private (FunctionInvokedEventArgs, FunctionResult) CallFunctionInvoked(Kernel kernel, SKContext context, FunctionResult result)
{
    var eventArgs = new FunctionInvokedEventArgs(this.GetMetadata(), result);
    if (kernel.OnFunctionInvoked(eventArgs))
    {
        // Apply any changes from the event handlers to final result.
        result = new FunctionResult(this.Name, eventArgs.SKContext, eventArgs.SKContext.Variables.Input)
        {
            // Updates the eventArgs metadata during invoked handler execution
            // will reflect in the result metadata
            Metadata = eventArgs.Metadata
        };
    }

    return (eventArgs, result);
}
```

Reply:
Changed to the pattern Stephen suggested